src/promptflow/tests/test_configs/tools/tool_with_enabled_by_value.py

```diff
@@ -0,0 +1,32 @@
+from enum import Enum
+
+from promptflow.entities import InputSetting
+from promptflow import tool
+
+
+class UserType(str, Enum):
+    STUDENT = "student"
+    TEACHER = "teacher"
+
+
+@tool(
+    name="My Tool with Enabled By Value",
+    description="This is my tool with enabled by value",
+    input_settings={"teacher_id": InputSetting(enabled_by="user_type", enabled_by_value=[UserType.TEACHER])}
+)
+def my_tool(user_type: Enum, student_id: str = "", teacher_id: str = "") -> str:
```

Comment:
Can we change the `student_id` also enabled_by user_type, enabled_by_value is UserType.Student ?

Reply:
Update the test case